modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotDeleteSafeZoneCalc.java

```diff
@@ -68,8 +68,13 @@ public void run() {
                 try {
                     nodesWithPublishedMessages = SlotManagerClusterMode.getInstance().getMessagePublishedNodes();
                 } catch (AndesException e) {
-                    log.error("SlotDeleteSafeZoneCalc stopped due to failing to get message published nodes.", e);
-                    this.setLive(false);
+                    log.error("SlotDeleteSafeZoneCalc stopped due to failing to get message published nodes. "
+                           + "Retrying after 15 seconds" ,e);
+                    try {
+                        Thread.sleep(15 * 1000);
+                    } catch (InterruptedException e1) {
+                        //ignore
```

Comment:
On interrupt should we ignore? [1]

[1] http://www.ibm.com/developerworks/library/j-jtp05236/